language/tools/cost-synthesis/src/bytecode_specifications/frame_transition_info.rs

```diff
@@ -23,13 +23,14 @@ pub(crate) fn frame_transitions<'txn>(
     instr: &Bytecode,
     module_info: (&'txn LoadedModule, Option<FunctionDefinitionIndex>),
 ) {
-    while interpreter.call_stack_height() > 1 {
+    while interpreter.call_stack_height() > 0 {
```

Comment:
not clear about those changes can you explain?

Reply:
It's tricky! But this is definitely needed: the call frame holds the current pc. If we reuse a frame from one call into the interpreter to another, we will fail to execute the instruction -- the instruction sequence will have length 1 but the pc in the frame will be greater than that. This is why we need to stop re-using the frame and instead push a brand new one.

Reply:
Thanks for the explanation! Yes, this is exactly what is causing the bytecodes to be run only in the first iteration.

Reply:
Perhaps an explanatory comment would be in order then?